client/src/components/users/UserProfile.tsx

```diff
@@ -0,0 +1,184 @@
+            value = Number.parseFloat(value);
+        }
+
+        // @ts-ignore
+        this.setState({
+            user: {
+                ...this.state.user,
+                [name]: value
+            }
+        });
+    }
+
+    public handleInputChangeCheckbox = (event: FormEvent<HTMLInputElement>, checkboxProps: CheckboxProps): void => {
+        if (checkboxProps && checkboxProps.name && typeof checkboxProps.checked !== "undefined") {
+            const value: boolean = checkboxProps.checked;
+            const name: string = checkboxProps.name;
+
+            // @ts-ignore
+            this.setState({
+                user: {
+                    ...this.state.user,
+                    [name]: value
+                }
+            });
+        }
+    };
+
+    public render() {
+        const isAdmin: boolean = !!this.props.user && this.props.user.admin;
+        const adminHeader = isAdmin ? <Header size="medium">User Administration</Header> : "";
+        const haveId = isAdmin ? <Popup inverted={true} trigger={<Form.Checkbox name="haveID"
+                                                                                label="Have ID"
+                                                                                checked={this.state.user.haveID}
+                                                                                onChange={this.handleInputChangeCheckbox}
+            />}
+                                        content="Whether the hardware desk currently has possession of this user's photo ID as collateral"/>
+            : "";
+        const userIsAdmin = isAdmin ? <Popup inverted={true} trigger={<Form.Checkbox label="Admin"
+                                                                                     name="admin"
+                                                                                     checked={this.state.user.admin}
+                                                                                     onChange={this.handleInputChangeCheckbox}
+            />}
+                                             content="Whether this user can access administration pages like this one"/>
+            : "";
+
+        const adminUuid = isAdmin ?
+            <Form.Field width={6}>
+                <label>UUID</label>
+                <p>{this.state.user.uuid}</p>
+            </Form.Field>
+            : "";
+
+        let yourThis = "your";
+        let pageTitle = "My Profile";
+        let aboutYou = "About You";
+
+        if (this.props.user && this.props.user.admin && this.props.user.uuid !== this.state.user.uuid) {
+            yourThis = "this";
+            pageTitle = `Edit ${this.state.user.name}`;
+            aboutYou = `About ${this.state.user.name}`;
+        }
+
+        return <div>
+            <Header size="huge">{pageTitle}</Header>
+            <Form loading={false}
+                  onChange={this.handleInputChange}
+                  error={!this.validateForm()}
```

Comment:
can we only run an error check on submit - it's intense to show error while people are still writing, esp when individual fields have errors as well